File: middleware/authMiddleware.js
import jwt from "jsonwebtoken";

export function requireAuth(req, res, next) {
  const authHeader = req.headers.authorization;
  if (!authHeader) return res.status(401).json({ error: "Missing Authorization header" });

  const [type, token] = authHeader.split(" ");
  if (type !== "Bearer" || !token) {
    return res.status(401).json({ error: "Invalid Authorization format" });
  }

  try {
    const decoded = jwt.verify(token, process.env.JWT_SECRET);
    req.user = {
      userId: decoded.userId,
      email: decoded.email,
      isAdmin: decoded.isAdmin
    };
    next();
  } catch {
    return res.status(401).json({ error: "Invalid or expired token" });
  }
}

File: middleware/auth_middleware.py
import os
import jwt
from fastapi import HTTPException, Security
from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer

security = HTTPBearer()

def require_auth(credentials: HTTPAuthorizationCredentials = Security(security)):
    """
    Dependency to require authentication.
    Validates JWT token and returns user data.
    """
    token = credentials.credentials
    
    if not token:
        raise HTTPException(
            status_code=401,
            detail="Missing Authorization header"
        )
    
    try:
        # Verify JWT token
        jwt_secret = os.getenv("JWT_SECRET")
        if not jwt_secret:
            raise HTTPException(
                status_code=500,
                detail="JWT_SECRET not configured"
            )
        
        decoded = jwt.decode(
            token,
            jwt_secret,
            algorithms=["HS256"]
        )
        
        # Return user data from token
        return {
            "userId": decoded.get("userId"),
            "correo": decoded.get("correo"),
            "nombre": decoded.get("nombre")
        }
    except jwt.ExpiredSignatureError:
        raise HTTPException(
            status_code=401,
            detail="Token has expired"
        )
    except jwt.InvalidTokenError:
        raise HTTPException(
            status_code=401,
            detail="Invalid or expired token"
        )
